### ISSUE TITLE
[dev]: Implement forward-compatibility for legacy nodes' ability to handle group-key-reveal changes 

### ISSUE THREAD

**Issue by dstadulis** [lightninglabs/taproot-assets#1251]
Given legacy nodes will encounter new validation data in https://github.com/lightninglabs/taproot-assets/pull/1246, there is a need to assess forwards compatibility of these nodes.

To endorse the implementation of the group-key reveal methods the following TODOs are necessary:

- [x] Assess compatibility of legacy nodes' ability to handle group-key-reveal changes
- [x] Create remediation to attempt to preserve forward-compatibility where reasonable

### TODO to preserve forward-compatibility
- [ ] Remove strict size check in current group key reveal parsing
- [ ] Create a new TLV type for the v1 group key reveal (even type)
  - [ ] Add rule that only v1 or v0 group key reveal can be set
  - [ ] Make value of this new field a nested TLV
- [ ] Consider adding compatibility adapter:
  - [ ] If new fields are present (new Merkle proofs), consider it v1; otherwise, consider it v0

Glossary:
GKR = Group key reveal

**Comment by ffranr** [comment]
If we modify the group key reveal field in proofs so that older nodes cannot decode it, syncing on older nodes may fail in a complex manner. This is because the group key reveal is used to determine the proof import order during universe syncing. Refer to:

https://github.com/lightninglabs/taproot-assets/blob/7358c1b0b42af8cf09d49942fb525ef367409d4f/universe/syncer.go#L306-L324

**Comment by ffranr** [comment]
TLDR:
* If GKR deocding is broken for older nodes then universe syncing might fail.
* If we don't populate the GKR field in a Proof then the proof won't be valid.

I guess that leaves the possibility of having a decodable dummy GKR and a new GKR TLV field in the Proof.

**Comment by Roasbeef** [comment]
Nice investigation! 

So I think it's a given that old nodes will fail to _verify_ the new group key reveal struct. IMO we still want to allow them to _parse_ the new modified field however. I think this points to us:
   * Removing the strict size check in the current group key reveal parsing. 
     * Either way, the older nodes will start to reject these new proofs as the encoding format will now differ. 
     * Similarly, if you can't verify this new group reveal, then you can't sync the universe root for the asset. 
       * This is isolated for all nodes other than those that are attempting to sync all issuance state all the time, based on the global multiverse root. 
   * Create a new TLV type for the v1 group key reveal. Make this an even type, as if it's set, then the parsing node needs to be able to handle it. 
      * Add a rule that only the v1 or v0 group key reveal can be set. 
      * We should then make the value of this new field a nested TLV. 
         * Optionally, we can make a sort of compatibility adapter here: if the new field are present (the new merkle proofs), then it's v1, otherwise it's v0.

**Comment by dstadulis** [comment]
Updated original comment with requested changes
